### DISCUSSION TITLE
Xvnc vs Xorg

### DISCUSSION THREAD

**Discussion by cwseys** [neutrinolabs/xrdp#2619] [General]
Hello all,
  It looks like Xorg is by default disabled while Xvnc is enabled.  I assume this means Xvnc is somehow preferred... Anyone know the pros and cons of running xrdp with Xorg versus Xvnc on the server side?

Thanks!
C.

**Comment by matt335672** [comment]
The preferred back-end depends on the distro. RHEL-based distros prefer Xvnc and Debian/Ubuntu prefer Xorg.

Off the top of my head:-

Xorg pros : Generally more performant as less data copying is involved. Can make use of some hardware accelerators.
Xorg cons : xorgxrdp uses Xorg internal interfaces which means it can be broken by system software or hardware changes.

Xvnc pros : Can work a little better than Xorg in some WAN configurations. In general, pretty bullet-proof.
Xvnc cons : Performance is noticeable worse than Xorg on a LAN.

**Reply by cwseys** [reply]
Thanks for all your help!
C.
